File: app/src/main/java/org/wikipedia/views/ConfigurableTextView.java
package org.wikipedia.views;

import android.content.Context;
import android.support.v7.widget.AppCompatTextView;
import android.text.Spanned;
import android.util.AttributeSet;

import static org.wikipedia.util.L10nUtil.setConditionalLayoutDirection;

public class ConfigurableTextView extends AppCompatTextView {
    public ConfigurableTextView(Context context) {
        super(context);
    }

    public ConfigurableTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ConfigurableTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setText(CharSequence text, String languageCode) {
        super.setText(text);
        setLocale(languageCode);
    }

    public void setText(Spanned text, String languageCode) {
        super.setText(text);
        setLocale(languageCode);
    }

    public void setLocale(String languageCode) {
        setConditionalLayoutDirection(this, languageCode);
    }
}

File: app/src/main/java/org/wikipedia/views/ConfigurableTextView.kt
package org.wikipedia.views

import android.content.Context
import android.util.AttributeSet
import com.google.android.material.textview.MaterialTextView
import org.wikipedia.util.L10nUtil

open class ConfigurableTextView constructor(context: Context, attrs: AttributeSet? = null) : MaterialTextView(context, attrs) {
    fun setText(text: CharSequence?, languageCode: String?) {
        super.setText(text)
        if (!isInEditMode) {
            L10nUtil.setConditionalLayoutDirection(this, languageCode!!)
        }
    }
}
